core/src/main/java/org/apache/iceberg/MetadataColumns.java

```diff
@@ -51,7 +51,7 @@ private MetadataColumns() {}
   public static final int SPEC_ID_COLUMN_ID = Integer.MAX_VALUE - 4;
   public static final String SPEC_ID_COLUMN_DOC = "Spec ID used to track the file containing a row";
   public static final NestedField SPEC_ID =
-      NestedField.required(
+      NestedField.optional(
```

Comment:
seems like an unrelated change?

Reply:
This change is needed in 4.0, check out https://github.com/apache/iceberg/pull/12494#discussion_r2040822469 and https://github.com/apache/spark/pull/50246#discussion_r2023825620

Reply:
There may be a more narrow way to do this rather than change it in core though, we can change in SparkTable when surfacing the metadata columns? cc @huaxingao

Reply:
Thanks for the suggestion. I have changed the metadata column optionality to Spark only.

Reply:
Thanks @huaxingao !